backend/src/main/java/com/redhat/cloud/notifications/routers/internal/DuplicateNameMigrationResource.java

```diff
@@ -1,161 +0,0 @@
-
-import com.redhat.cloud.notifications.auth.ConsoleIdentityProvider;
-import com.redhat.cloud.notifications.models.EndpointType;
-import com.redhat.cloud.notifications.routers.internal.models.DuplicateNameMigrationReport;
-import io.quarkus.logging.Log;
-import jakarta.annotation.PostConstruct;
-import jakarta.annotation.security.RolesAllowed;
-import jakarta.inject.Inject;
-import jakarta.persistence.EntityManager;
-import jakarta.transaction.Transactional;
-import jakarta.ws.rs.BadRequestException;
-import jakarta.ws.rs.GET;
-import jakarta.ws.rs.Path;
-import jakarta.ws.rs.Produces;
-import jakarta.ws.rs.QueryParam;
-
-import java.util.Arrays;
-import java.util.List;
-import java.util.Objects;
-import java.util.UUID;
-import java.util.stream.Collectors;
-
-import static com.redhat.cloud.notifications.Constants.API_INTERNAL;
-import static jakarta.ws.rs.core.MediaType.APPLICATION_JSON;
-
-@RolesAllowed(ConsoleIdentityProvider.RBAC_INTERNAL_ADMIN)
-@Path(API_INTERNAL + "/duplicate-name-migration")
-public class DuplicateNameMigrationResource {
-
-    final String ack = "i-am-sure-i-want-to-run-the-migration";
-
-    @Inject
-    EntityManager entityManager;
-
-    @PostConstruct
-    public void initialize() {
-        Log.infof("Duplicate name migration resource ack with value [%s]", ack);
-    }
-
-    @GET
-    @Produces(APPLICATION_JSON)
-    public DuplicateNameMigrationReport migrateDuplicateNames(@QueryParam("ack") String ack) {
-        if (!Objects.equals(this.ack, ack)) {
-            throw new BadRequestException(String.format("Invalid ack provided - should be [%s]", ack));
-        }
-
-        DuplicateNameMigrationReport migrationReport = new DuplicateNameMigrationReport();
-
-        Log.infof("Starting migration for duplicated names on behavior groups and integrations.");
-
-        while (true) {
-            DuplicateNameMigrationReport localReport = runMigrationStep();
-            migrationReport.updatedIntegrations += localReport.updatedIntegrations;
-            migrationReport.updatedBehaviorGroups += localReport.updatedBehaviorGroups;
-
-            if (localReport.updatedBehaviorGroups == 0 && localReport.updatedIntegrations == 0) {
-                break;
-            }
-        }
-
-        Log.infof("Finished updating duplicates names for behavior groups");
-
-        return migrationReport;
-    }
-
-    DuplicateNameMigrationReport runMigrationStep() {
-        DuplicateNameMigrationReport migrationReport = new DuplicateNameMigrationReport();
-
-        Log.infof("Running a migration step");
-
-        List<Object[]> repeatedIntegrationValues = entityManager.createNativeQuery(
-                        // string_agg concatenates all the grouped values with the delimiter (, in this case)
-                        "SELECT string_agg(CAST(id as character varying), ','), name FROM endpoints WHERE endpoint_type_v2 != :endpoint_type_email GROUP BY name, org_id HAVING count(*) > 1"
```

Comment:
Migration script does not cover internal endpoint names (email and drawer), must be updated before running it.